### ISSUE TITLE
InceptionResnetV2 comments

### ISSUE THREAD

**Issue by andrewgrahambrown** [Cadene/pretrained-models.pytorch#73]
Hi guys,

First thanks very much for the package! I found it very easy to use and am very grateful that you guys put in the effort to make this.

I was looking into the implementation of InceptionResnetv2 and comparing it to google's [description](https://ai.googleblog.com/2016/08/improving-inception-and-image.html). I see 3 areas where I can't quite match up the code to the online description, and I wondered what you thought about them. The 3 issues are annotated on this image:

![inceptionresnetv2_notes](https://user-images.githubusercontent.com/10993518/41404540-c690bc84-6fc7-11e8-914e-7bde3a8ec369.png)

1) Branch (1) in block 8 is three 2d convolutions rather than 2. I suspect this was on purpose given the kernel sizing of the second two convolutions [(1,3) then (3,1)] but I wanted to check.

2) I cannot find the dropout layer at the end of the network between avgpool_1a and last_linear   

3) I cannot see the extra branch that comes out of the middle of the network (between repeat/sequential/Block17 and mixed_7a in your implementation). I also guess this was on purpose, is this intended for the user to implement?

Thanks for your help!

**Comment by andrewgrahambrown** [comment]
1. The issue is that the lines 214/215 in inceptionresnetv2.py do not match Google's blog post [description](https://1.bp.blogspot.com/-O7AznVGY9js/V8cV_wKKsMI/AAAAAAAABKQ/maO7n2w3dT4Pkcmk7wgGqiSX5FUW2sfZgCLcB/s1600/image00.png) of InceptionResnetV2. However, lines 214/215 do match lines 92-95 in the TF implementation of [inception_resnet_v2.py](https://github.com/tensorflow/models/blob/master/research/slim/nets/inception_resnet_v2.py) and also match Fig. 13 in the related [publication](https://arxiv.org/pdf/1602.07261.pdf). So I have to suspect that it is Google's blog post that has a small error, not the code.

2. Happy to help there then!

3. I'm also not sure. I would suspect it causes a small difference if only because it is optional in the TF implementation.

**Comment by Cadene** [comment]
2. Feel free to submit a pull request ;)

3. Yeah I would expect it helps training with this branch. I am not sure if could find the time to add it tho. Why do you need it?

**Comment by andrewgrahambrown** [comment]
Make a fork with the dropout layer

**Comment by muzzart** [comment]
is there a fork with the dropout layer?
